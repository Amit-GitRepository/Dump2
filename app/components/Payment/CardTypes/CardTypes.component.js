/* CardTypes: Shows card type images serially*/

import jcb from '../../../assets/jcb.png';
import mastercard from '../../../assets/mastercard.png';
import React, {Component} from 'react';
import styles from './CardTypes.style';
import visa from '../../../assets/visa.png';
import {Image, View} from 'react-native';

class CardTypes extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Image source={visa} style={styles.image} />
        <Image source={mastercard} style={styles.image} />
        <Image source={jcb} style={[styles.image, styles.jcb]} />
      </View>
    );
  }
}

export default CardTypes;
