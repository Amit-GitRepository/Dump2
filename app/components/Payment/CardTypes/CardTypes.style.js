export default {
  container: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  image: {
    marginHorizontal: 5
  },
  jcb: {
    marginLeft: -5,
    bottom: 1
  }
};
