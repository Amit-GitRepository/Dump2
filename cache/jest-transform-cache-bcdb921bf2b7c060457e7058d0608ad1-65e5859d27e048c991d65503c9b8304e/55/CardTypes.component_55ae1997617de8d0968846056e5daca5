Object.defineProperty(exports,"__esModule",{value:true});var cov_cx80vbau4=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardTypes/CardTypes.component.js',hash='f4a737bf6a53989aad2ea95a551b9904e130490e',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardTypes/CardTypes.component.js',statementMap:{'0':{start:{line:12,column:4},end:{line:18,column:6}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:11,column:2},end:{line:11,column:3}},loc:{start:{line:11,column:12},end:{line:19,column:3}},line:11}},branchMap:{},s:{'0':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();

var _jcb=require('../../../assets/jcb.png');var _jcb2=_interopRequireDefault(_jcb);
var _mastercard=require('../../../assets/mastercard.png');var _mastercard2=_interopRequireDefault(_mastercard);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _CardTypes=require('./CardTypes.style');var _CardTypes2=_interopRequireDefault(_CardTypes);
var _visa=require('../../../assets/visa.png');var _visa2=_interopRequireDefault(_visa);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

CardTypes=function(_Component){_inherits(CardTypes,_Component);function CardTypes(){_classCallCheck(this,CardTypes);return _possibleConstructorReturn(this,(CardTypes.__proto__||Object.getPrototypeOf(CardTypes)).apply(this,arguments));}_createClass(CardTypes,[{key:'render',value:function render()
{cov_cx80vbau4.f[0]++;cov_cx80vbau4.s[0]++;
return(
_react2.default.createElement(_reactNative.View,{style:_CardTypes2.default.container},
_react2.default.createElement(_reactNative.Image,{source:_visa2.default,style:_CardTypes2.default.image}),
_react2.default.createElement(_reactNative.Image,{source:_mastercard2.default,style:_CardTypes2.default.image}),
_react2.default.createElement(_reactNative.Image,{source:_jcb2.default,style:[_CardTypes2.default.image,_CardTypes2.default.jcb]})));


}}]);return CardTypes;}(_react.Component);exports.default=


CardTypes;