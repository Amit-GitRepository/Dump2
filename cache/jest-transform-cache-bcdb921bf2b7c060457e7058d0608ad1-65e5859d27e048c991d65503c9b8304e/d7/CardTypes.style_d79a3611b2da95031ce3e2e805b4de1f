Object.defineProperty(exports,"__esModule",{value:true});var cov_22pxas1rok=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardTypes/CardTypes.style.js',hash='cdd228a75d68b0adf7cb09189ffebb83649e4679',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Payment/CardTypes/CardTypes.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();exports.default={
container:{
flexDirection:'row',
alignItems:'center'},

image:{
marginHorizontal:5},

jcb:{
marginLeft:-5,
bottom:1}};