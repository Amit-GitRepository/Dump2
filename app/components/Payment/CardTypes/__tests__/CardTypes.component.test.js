import CardTypes from '../CardTypes.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('CardTypes component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<CardTypes />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
