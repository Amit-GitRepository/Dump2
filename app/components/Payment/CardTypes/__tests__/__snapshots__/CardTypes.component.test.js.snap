// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTypes component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/visa.png",
      }
    }
    style={
      Object {
        "marginHorizontal": 5,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/mastercard.png",
      }
    }
    style={
      Object {
        "marginHorizontal": 5,
      }
    }
  />
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/jcb.png",
      }
    }
    style={
      Array [
        Object {
          "marginHorizontal": 5,
        },
        Object {
          "bottom": 1,
          "marginLeft": -5,
        },
      ]
    }
  />
</View>
`;
